sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/AlwaysFalse.java

```diff
@@ -17,34 +17,32 @@
 
 package org.apache.spark.sql.connector.expressions.filter;
 
-import java.util.Objects;
-
 import org.apache.spark.annotation.Evolving;
-import org.apache.spark.sql.connector.expressions.NamedReference;
+import org.apache.spark.sql.connector.expressions.Literal;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
 
 /**
- * A filter that always evaluates to {@code false}.
+ * A predicate that always evaluates to {@code false}.
  *
  * @since 3.3.0
  */
 @Evolving
-public final class AlwaysFalse extends Filter {
+public class AlwaysFalse extends Predicate implements Literal<Boolean> {
```

Comment:
Does this need to implement `Literal`?

Reply:
Based on the suggestion https://github.com/apache/spark/pull/35768#discussion_r824523932

Reply:
This allows people to use different ways to match `AlwaysFalse`